src/cli/logs.ts

```diff
@@ -28,13 +49,29 @@ export const startupLogVersion = (logger: Logger, task: TaskCommand, coreCompile
   logger.info(startupMsg);
 };
 
+/**
+ * Log details from a {@link CompilerSystem} used by Stencil to an output stream
+ *
+ * The output stream is determined by the {@link Logger} instances that is provided as an argument to this function
+ *
+ * @param sys the `CompilerSystem` to report details on
+ * @param logger the logging entity to use for output
+ * @param flags user set flags for the current invocation of Stencil
+ * @param coreCompiler the compiler instance being used for this invocation of Stencil
+ */
 export const loadedCompilerLog = (
   sys: CompilerSystem,
   logger: Logger,
   flags: ConfigFlags,
   coreCompiler: CoreCompiler
-) => {
-  const sysDetails = sys.details;
+): void => {
+  const sysDetails: SystemDetails = sys.details ?? {
+    platform: '',
+    cpuModel: 'UNKNOWN',
+    freemem: () => 0,
+    release: 'UNKNOWN',
+    totalmem: 0,
+  };
```

Comment:
Technically this is a change in functionality. However, prior to this change, `sysDetails` could be `undefined`, and users would run into "trying to access [PROPERTY_NAME] of undefined" errors on `run()`

Reply:
Does it make sense for these logs to print if there are no `sysDetails`? Should another message be printed instead?

Also, for my knowledge, in what case would the details not exist?

Reply:
> Does it make sense for these logs to print if there are no sysDetails? Should another message be printed instead?

Fair, we could alternatively log something like "Unable to determine System Details"

> Also, for my knowledge, in what case would the details not exist?

Someone could invoke `run()` manually via the Node API, where they handcraft the `sys` entity :-(

Reply:
> Fair, we could alternatively log something like "Unable to determine System Details"

I'd be in favor of something like this instead. Feels clearer than us generating "fake" information.

Reply:
👍 SGTM - I'll wait for @alicewriteswrongs to weigh in and make those changes once we have consensus

Reply:
thanks all - added https://github.com/ionic-team/stencil/pull/3589/commits/bf69f45b9f46f7a516e21978ab1a9c8f3a3b14c8 in order to keep similar behavior for cases where the logger is set to debug or `flags.CI` is truthy, without introducing a separate conditional. LMK what yinz think